Objects/bytearrayobject.c

```diff
@@ -267,6 +267,7 @@ PyByteArray_Concat(PyObject *a, PyObject *b)
     result = (PyByteArrayObject *) \
         PyByteArray_FromStringAndSize(NULL, va.len + vb.len);
     if (result != NULL) {
+        assert(result->ob_bytes != NULL);
```

Comment:
The two memcpy can be skipped if result length is zero:

```suggestion
    // result->ob_bytes is NULL if result is an empty string:
    // if va.len + vb.len equals zero.
    if (result != NULL && result->ob_bytes != NULL) {
```

IMHO it's ok to call memcpy(dst, src, 0) if src and dst are non-NULL: it's not an undefined behavior (if I understood correctly).

Reply:
Rebased with this suggestion. With this fix the error is not reported any more by the static analysis tool.